interfaces/cython/cantera/cti2yaml.py

```diff
@@ -0,0 +1,1648 @@
+        :param id:
+            An optional identification string.
+        :param order:
+            Override the default reaction orders implied by the reactant
+            stoichiometric coefficients. Given as a string of key:value pairs,
+            e.g., ``"CH4:0.25 O2:1.5"``.
+        :param options: Processing options, as described in
+            `Options <https://cantera.org/tutorials/cti/reactions.html#options>`_.
+            May be one or more (as a list) of the
+            following: ``'duplicate'``, ``'negative_A'``,`` 'negative_orders'``,
+            ``'nonreactant_orders'``.
+        """
+        self._e = equation
+        self._order = getComposition(order)
+        self._num = len(_reactions['reactions']) + 1
+        self._id = id
+        self._options = [options] if isinstance(options, str) else options
+        self._kf = Arrhenius(*kf) if isinstance(kf, (list, tuple)) else kf
+        self._type = 'elementary'
+        _reactions['reactions'].append(self)
+
+    @classmethod
+    def to_yaml(cls, representer, node):
+        out = BlockMap()
+        node.get_yaml(out)
+        return representer.represent_dict(out)
+
+    def get_yaml(self, out):
+        out['equation'] = self._e
+        out.yaml_add_eol_comment('Reaction {}'.format(self._num), 'equation')
+        if self._type not in ('elementary', 'edge', 'surface'):
+            out['type'] = self._type
+
+        if self._id:
+            out['id'] = self._id
+
+        if self._type in ('elementary', 'three-body', 'edge', 'surface'):
+            out['rate-constant'] = self._kf
+
+        if 'duplicate' in self._options:
+            out['duplicate'] = True
+        if 'negative_A' in self._options:
+            out['negative-A'] = True
+
+        if self._order:
+            out['orders'] = FlowMap(self._order.items())
+        if 'negative_orders' in self._options:
+            out['negative-orders'] = True
+        if 'nonreactant_orders' in self._options:
+            out['nonreactant-orders'] = True
+
+
+class three_body_reaction(reaction):
+    """
+    A three-body reaction.
+    """
+    def __init__(self, equation, kf, efficiencies='', id='', options=[]):
+        """
+        :param equation:
+            A string specifying the chemical equation. The reaction can be
+            written in either the association or dissociation directions, and
+            may be reversible or irreversible.
+        :param kf:
+            The rate coefficient for the forward direction. If a sequence of
+            three numbers is given, these will be interpreted as [A, b, E] in
+            the modified Arrhenius function.
+        :param efficiencies:
+            A string specifying the third-body collision efficiencies.
+            The efficiencies for unspecified species are set to 1.0.
+        :param id:
+            An optional identification string. If omitted, it defaults to a
+            four-digit numeric string beginning with 0001 for the first
```

Comment:
Well, in `ctml_writer`, it will just print more digits as necessary, i.e. reactions up to 999 will be prefixed with zeros, but the ones after that will not.

But here, this is actually just a mistake in the documentation -- `cti2ctml` doesn't output an `id` attribute for a reaction unless you specify one explicitly.